Parser/pegen.c

```diff
@@ -681,11 +681,46 @@ initialize_token(Parser *p, Token *token, const char *start, const char *end, in
         return -1;
     }
 
-    const char *line_start = token_type == STRING ? p->tok->multi_line_start : p->tok->line_start;
-    int lineno = token_type == STRING ? p->tok->first_lineno : p->tok->lineno;
+    const char *line_start;
+    if (token_type == STRING) {
+        line_start = p->tok->multi_line_start;
+    }
+    else if (token_type == FSTRING_MIDDLE) {
+        line_start = p->tok->tok_mode_stack[p->tok->tok_mode_stack_index].f_string_constant_multi_line_start;
+    }
+    else if (token_type == FSTRING_END) {
+        line_start = p->tok->prev_tokenizer_stack_item->f_string_constant_multi_line_start;
+    }
```

Comment:
For these two, can we somehow assert that `prev_tokenizer_stack_item` is never going to be null?